tests/login.e2e.ts

```diff
@@ -36,18 +41,18 @@ test('Successful login for teacher', async (t) => {
 });
 test('Failed login for student', async (t) => {
   await loginPage.loginAsStudent();
-  await assertLoginFailure(t);
+  await assertOnLoginPage(t);
 });
 
 test('Failed login - wrong credentials', async (t) => {
   // wrong password
-  await loginPage.login('teacher', 'wrong-password');
+  await loginPage.login(teacherUser, 'wrong-password');
   await t.expect(loginPage.wrongCredentialsErr.exists).ok();
-  await assertLoginFailure(t);
```

Comment:
expect is chainable please use chains:
```
  await t
  .expect(loginPage.wrongCredentialsErr.exists)
  .ok()
  .expect(location.pathname).notContains('student')
  .expect(location.pathname).contains('login');
```

explicit is better than implicit
https://www.python.org/dev/peps/pep-0020/